core/rpc/src/lib.rs

```diff
@@ -34,13 +36,16 @@ static HMAC_SECRET: OnceLock<[u8; 32]> = OnceLock::new();
 pub static HMAC_NONCE: AtomicUsize = AtomicUsize::new(0);
 pub static HMAC_SALT: &[u8] = b"lightning-hmac-salt";
 
-/// Tries to read the hmac secret from the default FS location
+/// Tries to read the hmac secret from the given path or the default FS location
 /// or it will generate one and write it to disk
-pub fn hmac_secret() -> anyhow::Result<&'static [u8; 32]> {
+pub fn hmac_secret(secret_dir_path: Option<PathBuf>) -> anyhow::Result<&'static [u8; 32]> {
     match HMAC_SECRET.get() {
         Some(secret) => Ok(secret),
         None => {
-            let parent = ResolvedPathBuf::try_from(LIGHTNING_HOME_DIR.clone())?;
+            let path = secret_dir_path.unwrap_or_else(|| LIGHTNING_HOME_DIR.to_path_buf());
```

Comment:
@nhtyy I can reproduce the failing test locally by deleting `~/.lightning` before running it. When I add `RUST_BACKTRACE=1` I can see where it's coming from (line 72 in here); seems to be still trying to create the file in `LIGHTNING_HOME`